orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java

```diff
@@ -6,260 +6,165 @@
  */
 package org.hibernate.datastore.ogm.orientdb;
 
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.SQLException;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-
-import org.hibernate.datastore.ogm.orientdb.dialect.impl.OrientDBAssociationQueries;
-import org.hibernate.datastore.ogm.orientdb.dialect.impl.OrientDBEntityQueries;
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.id.ORecordId;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
+import org.hibernate.datastore.ogm.orientdb.dialect.impl.OrientDBQueryHelper;
 import org.hibernate.datastore.ogm.orientdb.dialect.impl.OrientDBTupleSnapshot;
 import org.hibernate.datastore.ogm.orientdb.impl.OrientDBDatastoreProvider;
 import org.hibernate.datastore.ogm.orientdb.logging.impl.Log;
 import org.hibernate.datastore.ogm.orientdb.logging.impl.LoggerFactory;
 import org.hibernate.datastore.ogm.orientdb.type.spi.ORecordIdGridType;
-import org.hibernate.engine.spi.SessionFactoryImplementor;
-import org.hibernate.ogm.dialect.identity.spi.IdentityColumnAwareGridDialect;
-import org.hibernate.ogm.dialect.spi.AssociationContext;
-import org.hibernate.ogm.dialect.spi.AssociationTypeContext;
-import org.hibernate.ogm.dialect.spi.BaseGridDialect;
-import org.hibernate.ogm.dialect.spi.ModelConsumer;
-import org.hibernate.ogm.dialect.spi.NextValueRequest;
-import org.hibernate.ogm.dialect.spi.SessionFactoryLifecycleAwareDialect;
-import org.hibernate.ogm.dialect.spi.TupleAlreadyExistsException;
-import org.hibernate.ogm.dialect.spi.TupleContext;
+import org.hibernate.ogm.dialect.batch.spi.OperationsQueue;
+import org.hibernate.ogm.dialect.spi.*;
 import org.hibernate.ogm.model.key.spi.AssociationKey;
 import org.hibernate.ogm.model.key.spi.AssociationKeyMetadata;
 import org.hibernate.ogm.model.key.spi.EntityKey;
 import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
 import org.hibernate.ogm.model.spi.Association;
 import org.hibernate.ogm.model.spi.Tuple;
-import org.hibernate.ogm.persister.impl.OgmCollectionPersister;
-import org.hibernate.ogm.persister.impl.OgmEntityPersister;
+import org.hibernate.ogm.model.spi.TupleOperation;
 import org.hibernate.ogm.type.spi.GridType;
-import org.hibernate.persister.collection.CollectionPersister;
-import org.hibernate.persister.entity.EntityPersister;
 import org.hibernate.type.Type;
 
-import com.orientechnologies.orient.core.id.ORecordId;
+import java.util.Set;
 
 /**
  * @author Sergey Chernolyas (sergey.chernolyas@gmail.com)
+ * @author cristhiank (calovi86 at gmail.com)
  */
-public class OrientDBDialect extends BaseGridDialect
-		implements SessionFactoryLifecycleAwareDialect, IdentityColumnAwareGridDialect {
```

Comment:
I did it because the 2.1.x versions of OrientDB doesn't support identity or autoincrement column generation, it's supported since 2.2.x, I'm not sure if this is right but that's why I removed it.
See http://orientdb.com/docs/2.1/Sequences-and-auto-increment.html